File: 0162-find-peak-element/0162-find-peak-element.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        n = len(nums)
        left,right = 0,n-1
        
        while left < right:
            mid = left + (right-left)//2
            
            if mid < n-1 and nums[mid] > nums[mid+1]:
                right = mid
            else:
                left = mid+1
            
            
        
        return left
            
            
        
        
        
        

File: 0162-find-peak-element/0162-find-peak-element.java
class Solution {
    public int findPeakElement(int[] nums) {
        
        int left = 0, right = nums.length-1;
        
        while(left < right){
            int mid = left + (right-left)/2;
            
            if(mid < nums.length-1 && nums[mid] > nums[mid+1]) right = mid;
            else left = mid+1;
        }
        
        return left;
        
    }
}

/*

1,2,1,3,5,6,4
        m
          i
          j


*/